### PR TITLE
Honor -nostdlib flag

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2214,7 +2214,8 @@ def compile_source_file(i, input_file):
          not shared.Settings.BOOTSTRAPPING_STRUCT_INFO and \
          not shared.Settings.SIDE_MODULE: # shared libraries/side modules link no C libraries, need them in parent
         extra_files_to_link = system_libs.get_ports(shared.Settings)
-        extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, in_temp, stdout_=None, stderr_=None, forced=forced_stdlibs)
+        if '-nostdlib' not in newargs and '-nodefaultlibs' not in newargs:
```
Comment (kripken) [inline emcc.py]: should this affect ports as well on the previous line? if so it can be part of that `if`

File: emcc.py [hunk 2]
```diff
@@ -3749,7 +3750,9 @@ def process_libraries(libs, lib_dirs, temp_files):
   # Find library files
   for i, lib in libs:
     logger.debug('looking for library "%s"', lib)
-    suffixes = STATICLIB_ENDINGS + DYNAMICLIB_ENDINGS
+    suffixes = list(STATICLIB_ENDINGS + DYNAMICLIB_ENDINGS)
+    if not shared.Settings.WASM_BACKEND:
+      suffixes.append('.bc')
```
Comment (kripken) [inline emcc.py]: this seems unrelated, or I'm missing the connection?

File: tests/test_other.py [hunk 3]
```diff
@@ -10552,3 +10552,17 @@ def test_bitcode_input(self):
...
+
+    self.assertContained('undefined symbol:', self.expect_fail([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nostdlib']))
+    self.assertContained('undefined symbol:', self.expect_fail([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nodefaultlibs']))
+
+    # Buil again but with explit system libraries
+    libs = ['-lc', '-lcompiler_rt']
+    if self.is_wasm_backend():
+      libs.append('-lc_rt_wasm')
+    run_process([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nostdlib'] + libs)
+    run_process([PYTHON, EMCC, path_from_root('tests', 'unistd', 'close.c'), '-nodefaultlibs'] + libs)
```
Comment (kripken) [inline tests/test_other.py]: can have one big loop on the outside for the two `-no*` flags, to avoid 2 sets of duplicate lines?

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py]: I don't think it should effect ports.    I think it should effect system libraries only.

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py]: This allows `-lc` to work on the command line for  for asmjs librarys such as libc.bc.

File: tests/test_other.py (hunk 3 above)
Comment (sbc100) [reply tests/test_other.py]: I have an aversion to loops like that in test code since it makes failure messages harder to interpret.     if it work many duplicate lines I might agree with you, and make parenthesized test.  In this case I'm ok with the duplication level.

Comment (kripken) [reply tests/test_other.py]: Sounds good.
